### ISSUE TITLE
Updating examples using react-router-dom v6

### ISSUE THREAD

**Issue by duzgundev** [reduxjs/redux-toolkit#2134]
Hi,

I've searched through all issues (including closed ones) and couldn't find anything about this topic. I love using RTK and RTK Query and I think the examples section is great. But they're still using `react-router-dom` v5. Since I recommend this library to several people, this situation is a little bump in the road according to feedbacks I receive. I would love to update all the examples using `react-router-dom` v6 if there's no planned action on that topic.

Thanks.

**Comment by phryneas** [comment]
I mean, of course PRs against the examples are always welcome - but is that in any way really relevant on how to use RTK Query?

**Comment by duzgundev** [comment]
Not really. Actullay, RTK Query examples are so good for learning concepts like authentication. But react router has changed a lot and some people are getting frustrated while they try to build their own apps using RTK Query via following examples. What I'm trying to say is, it could be better for beginners since they won't have to figure out routing on their own. That's not a crucial thing, it targets beginners mostly.

**Comment by markerikson** [comment]
Yeah, as Lenz said, PRs are definitely welcome. I _am_ curious - got any pointers to where "people getting frustrated", as far as discussion?

**Comment by duzgundev** [comment]
For example, the `PrivateRoute.tsx` component doesn't work with v6. I know the equivalent methods are written in react-router docs but this means an extra step for a person who wants to implement basic auth for instance. Maybe the term 'frustrated' is a little bit too much :)

**Comment by pauljosephatay** [comment]
and the new react router v6 data api aka loader too although i think react router loader and RTK are not compatible design wise.
can't use hooks in loaders

**Comment by markerikson** [comment]
@pauljosephatay : see discussion of RTK + the React-Router loaders in https://github.com/reduxjs/redux-toolkit/discussions/2751 .

To be frank, we don't have time or energy to update the routing examples ourselves.  We'd happily accept PRs that improve things!

**Comment by pauljosephatay** [comment]
Thanks a ton @markerikson for the link!

**Comment by seanmcquaid** [comment]
Any objection from anyone if I work on a PR for this one?

**Comment by markerikson** [comment]
@seanmcquaid : please, go for it!

**Comment by Adagedo** [comment]
I have noticed the last update on this issue is roughly two years now. Any objection if I update the existing PR?

**Comment by seanmcquaid** [comment]
No objection on my end!
